packages/modules/Extension-Ads/src/worker/worker.js

```diff
@@ -11,16 +11,21 @@ const extensionAdsAppMgr = {
 
     chrome.tabs.onRemoved.addListener((tabId) => delete this.tabData[tabId]);
 
-    chrome.tabs.onUpdated.addListener((tabId, changeInfo, { title, url, status }) => {
-      // create or update tabData
+    chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+      const { title, url, status } = tab;
+
+       // create or update tabData
       if (!this.tabData[tabId]) {
-        this.tabData[tabId] = { ads: new Map(), title, url, status };
+        this.tabData[tabId] = { ads: new Map(), title, url, status, prevUrl: null };
       } else if (status === 'loading') {
+        if (this.tabData[tabId].url !== url) {
```

Comment:
Okay, but what happens if a user clicks on a URL with target="_blank"? 
New tab with new tabId will be opened.